src/components/i18n/FormattedCompMessage.js

```diff
@@ -0,0 +1,187 @@
+                }
+            } else {
+                source = composition.compose();
+            }
+
+            this.state = {
+                id,
+                source,
+                composition,
+            };
+        }
+    }
+
+    /**
+     * Search for any Plural elements in the children, and
+     * then construct the English source string in the correct
+     * format for react-intl to use for pluralization
+     * @param {React.Element} children the children of this node
+     * @return {string} the composed plural string
+     */
+    composePluralString(children) {
+        const categories = {};
+        React.Children.forEach(children, child => {
+            if (typeof child === 'object' && React.isValidElement(child) && child.type.name === 'Plural') {
+                const childComposition = new Composition(child.props.children);
+                categories[child.props.category] = childComposition.compose();
+            }
+        });
+        if (!categories.one || !categories.other) {
+            if (isDevEnvironment()) {
+                throw new Error(
+                    'Cannot use count prop on a FormattedCompMessage component without giving both a "one" and "other" Plural component in the children.',
+                );
+            }
+        }
+        // add these to the string in a particular order so that
+        // we always end up with the same string regardless of
+        // the order that the Plural elements were specified in
+        // the source code
+        const categoriesString = [
+            CATEGORY_ZERO,
+            CATEGORY_ONE,
+            CATEGORY_TWO,
+            CATEGORY_FEW,
+            CATEGORY_MANY,
+            CATEGORY_OTHER,
+        ]
+            .map(category => (categories[category] ? ` ${category} {${categories[category]}}` : ''))
+            .join('');
+
+        // see the intl-messageformat project for an explanation of this syntax
+        return `{count, plural,${categoriesString}}`;
+    }
+
+    render() {
+        const { count, tagName, intl, description } = this.props;
+        const { composition, id, source } = this.state;
+        const values = {};
+        if (typeof count === 'number') {
+            // make sure intl.formatMessage switches properly on the count
+            values.count = count;
+        }
+
+        // react-intl will do the correct plurals if necessary
+        const translation = intl.formatMessage(
+            {
+                id,
+                defaultMessage: source,
+                description,
+            },
+            values,
+        );
+
+        // always wrap the translated string in a tag to contain everything
+        // and to give us a spot to record the id
+        return React.createElement(
+            tagName,
+            {
+                key: id,
+                'x-resource-id': id,
+            },
```

Comment:
Would it be worthwhile to spread any other props onto the inner span? That way we could apply a className to it directly, for example.